File: avalon-support/avalon-oauth2-support/avalon-oauth2-common/src/test/java/io/github/lishangbu/avalon/oauth2/common/log/AuthenticationLogTests.java
package io.github.lishangbu.avalon.oauth2.common.log;

import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertNull;
import static org.junit.jupiter.api.Assertions.assertFalse;

import java.time.Instant;
import org.junit.jupiter.api.Test;

class AuthenticationLogTests {

    @Test
    void recordStoresProvidedValues() {
        Instant now = Instant.now();
        AuthenticationLogRecord record =
                new AuthenticationLogRecord("alice", "client", "password", "127.0.0.1", "UA", true, "none", now);

        assertEquals("alice", record.username());
        assertEquals("client", record.clientId());
        assertEquals("password", record.grantType());
        assertEquals("127.0.0.1", record.ip());
        assertEquals("UA", record.userAgent());
        assertEquals(true, record.success());
        assertEquals("none", record.errorMessage());
        assertEquals(now, record.timestamp());
    }

    @Test
    void recordSupportsNullsAndFalseFlag() {
        AuthenticationLogRecord record =
                new AuthenticationLogRecord(null, null, null, null, null, false, null, null);

        assertNull(record.username());
        assertNull(record.clientId());
        assertNull(record.grantType());
        assertNull(record.ip());
        assertNull(record.userAgent());
        assertFalse(record.success());
        assertNull(record.errorMessage());
        assertNull(record.timestamp());
    }

    @Test
    void noopRecorderAcceptsRecords() {
        AuthenticationLogRecorder recorder = AuthenticationLogRecorder.noop();

        assertDoesNotThrow(() -> recorder.record(new AuthenticationLogRecord(null, null, null, null, null, false, null, null)));
    }
}

File: avalon-platform/avalon-security/avalon-oauth2-common/src/test/kotlin/io/github/lishangbu/avalon/oauth2/common/log/AuthenticationLogTests.kt
package io.github.lishangbu.avalon.oauth2.common.log

import org.junit.jupiter.api.Assertions.*
import org.junit.jupiter.api.Test
import java.time.Instant

class AuthenticationLogTests {
    @Test
    fun recordStoresProvidedValues() {
        val now = Instant.now()
        val record =
            AuthenticationLogRecord(
                "alice",
                "client",
                "password",
                "127.0.0.1",
                "UA",
                true,
                "none",
                now,
            )

        assertEquals("alice", record.username)
        assertEquals("client", record.clientId)
        assertEquals("password", record.grantType)
        assertEquals("127.0.0.1", record.ip)
        assertEquals("UA", record.userAgent)
        assertEquals(true, record.success)
        assertEquals("none", record.errorMessage)
        assertEquals(now, record.timestamp)
    }

    @Test
    fun recordSupportsNullsAndFalseFlag() {
        val record = AuthenticationLogRecord(null, null, null, null, null, false, null, null)

        assertNull(record.username)
        assertNull(record.clientId)
        assertNull(record.grantType)
        assertNull(record.ip)
        assertNull(record.userAgent)
        assertFalse(record.success)
        assertNull(record.errorMessage)
        assertNull(record.timestamp)
    }

    @Test
    fun noopRecorderAcceptsRecords() {
        val recorder = AuthenticationLogRecorder.noop()

        assertDoesNotThrow {
            recorder.record(
                AuthenticationLogRecord(null, null, null, null, null, false, null, null),
            )
        }
    }
}
